src/reconcile.rs

```diff
@@ -285,6 +317,226 @@ where
         }
     }
 
+    /// This method can be used to ensure a ConfigMap exists and has the specified content.
+    ///
+    /// If a ConfigMap with the specified name does not exist it will be created.
+    ///
+    /// Should a ConfigMap with the specified name already exist the content is retrieved and
+    /// compared with the content from `config_map`, if content differs the existing ConfigMap is
+    /// updated.
+    ///
+    /// Returns `Ok(true)` if a change was made and `Ok(false}` if no change was necessary.
+    pub async fn create_config_map(&self, config_map: ConfigMap) -> OperatorResult<bool> {
+        let cm_name = match config_map.metadata.name.as_deref() {
+            None => {
+                return Err(InvalidName {
+                    errors: vec![String::from(
+                        "ConfigMap with empty name encountered, this is illegal!",
+                    )],
+                })
+            }
+            Some(name) => name,
+        };
+
+        match self
+            .client
+            .get::<ConfigMap>(cm_name, Some(&self.namespace()))
+            .await
+        {
+            Ok(ConfigMap {
+                data: existing_config_map_data,
+                ..
+            }) if existing_config_map_data == config_map.data => {
+                info!(
+                    "ConfigMap [{}] already exists with identical data, skipping creation!",
+                    cm_name
+                );
+                Ok(false)
+            }
+            Ok(_) => {
+                info!(
+                    "ConfigMap [{}] already exists, but differs, updating it!",
+                    cm_name
+                );
+                self.client.update(&config_map).await?;
+                Ok(true)
+            }
+            Err(KubeError {
+                source: kube::error::Error::Api(ErrorResponse { reason, .. }),
+            }) if reason == "NotFound" => {
+                info!("ConfigMap [{}] not found, creating it.", cm_name);
+                self.client.create(&config_map).await?;
+                Ok(true)
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    pub async fn default_restart<C, P>(
+        &self,
+        command: &C,
+        pod_provider: &P,
+    ) -> ReconcileResult<Error>
+    where
+        T: HasApplication + HasRoleRestartOrder,
+        C: Command + CanBeRolling + HasRoles,
+        // TODO: Not sure if we can skip 'HasRoles' here and conditinally run code below if it is implemented
```

Comment:
I dont think that works. You can implement it "conditionally" via impl blocks with different where clauses. But that wont safe much code. 
But i think we have to differentiate. You cannot call this method now if no HasRoles is implemented?